src/circuit_builder.rs

```diff
@@ -190,56 +134,52 @@ where
         &mut self,
         pub_params: &PublicParameters,
         verifier_key: &VerifierKey,
-        transcript_initialisation: &'static [u8],
         proof: &Proof,
-        pub_inputs: &[PublicInput],
+        pub_inputs_values: &[PublicInputValue],
+        pub_inputs_positions: &PublicInputPositions,
```

Comment:
As mentioned in the comment related to `PublicInputPosition`:
```suggestion
        pub_input_positions: &[usize],
```